### ISSUE TITLE
"ghost blocks" prevent player from placing down blocks in gaia dimension

### ISSUE THREAD

**Issue by MasterEnderman** [Lemonszz/Corvus#6]
Please fill out this template when submitting an issue.

---
**Mod Version**:  1.2.3

**Forge Version**: 14.23.5.2838

**This issue occurs on the lastest version of Forge? (Y/N)**: Y

**What happens**:

In the gaia dimension there're randomly generated "blocks" that prevent placing down blocks in these locations. It only happens in this dimension. And there's no way to get rid of them.

example: https://streamable.com/rzeyb 
It also prevents mobs from passing through: https://streamable.com/fy86q

**Comment by Lemonszz** [comment]
I have literally no clue what that could be.. interesting.

**Comment by Lemonszz** [comment]
Can you recreate this with just Corvus installed?

**Comment by hurja** [comment]
since the world in question is mine ill add to the confusion.

blocks that can be placed in the offending spots include but arent limited to:
vanilla torch
water and lava
saplings
maybe others?
https://streamable.com/k0hv6

also the offending area seems to be 2 blocks high. third block is fine to place blocks in
https://streamable.com/ocaeu

in new worlds these spots are much more uncommon than in the world where the videos are from. i have found small 1-2 block sized areas in new worlds. the world in my videos is several hours old so time may be a factor.

**Comment by Lemonszz** [comment]
I'll look into it but I can't guarantee anything as I'm fairly sure this must be some weird mod interaction.

Are you 100% sure it only happens in the Forest of Gaia?

**Comment by MasterEnderman** [comment]
it couldn't be reproduced in any other dimension yet.

**Comment by hurja** [comment]
seems as time passes the dimension somehow corrupts more.

less than an hour old gaia dimension. holes in stone are spots block cant be placed
https://i.imgur.com/xegf2ad.png

over 9 hours old dimension https://streamable.com/ym05g
and it is really stuttery and jerky for no apparent reason.

**Comment by hurja** [comment]
`/kill @e[type=!player,r=3]` produces something interesting? https://streamable.com/xtf27

it occured me to blow stuff up https://streamable.com/kvj59

huh.

**Comment by Lemonszz** [comment]
Looks like a bunch of Witch drops?

Could you try the /kill command again and send me the log.
I still have no clue.

If anyone else can recreate this in a different modpack, that would help a lot.

**Comment by hurja** [comment]
https://www.dropbox.com/s/l9urg8mokrkmskr/latest.log?dl=0
https://www.dropbox.com/s/wfw7cf6n1vejigq/debug.log?dl=0

these should be viewable? both were from same pack load, /kill, exit game.

**Comment by taint19711** [comment]
what launcher are you using hurja

**Comment by taint19711** [comment]
im having the same issue using the twitch launcher

**Comment by Lemonszz** [comment]
What launcher you're using wont affect anything.

It seems to be a conflict with some unknown mod.
If anyone could recreate this with JUST Corvus/Lemonlib installed or find which mod is conflicting, that would be great.

**Comment by vertex101** [comment]
would anyone be able to provide a world that is effected by the ghost blocks to be able to track down the issue ?

**Comment by hurja** [comment]
affected test world https://mega.nz/#!QCAwTS4b!CVDYeNYdhhHylvPjwVBXChv5YvcvJ6GlSkQN566JOko

modpack version is Divine-Equilibrium-0.2.0-pre5 if you are going to start it in minecraft in MasterEndermans github.

**Comment by vertex101** [comment]
> affected test world https://mega.nz/#!QCAwTS4b!CVDYeNYdhhHylvPjwVBXChv5YvcvJ6GlSkQN566JOko
> 
> modpack version is Divine-Equilibrium-0.2.0-pre5 if you are going to start it in minecraft in MasterEndermans github.

yeah i done some testing and i loaded up that world into MCEdit and there is over 1700 witches in the gaia dim there is like 3 to 4 stacked on top of each other 

https://i.imgur.com/ZH3WqTX.gifv

i am still trying to figure out how they are getting there

**Comment by vertex101** [comment]
> What launcher you're using wont affect anything.
> 
> It seems to be a conflict with some unknown mod.
> If anyone could recreate this with JUST Corvus/Lemonlib installed or find which mod is conflicting, that would be great.

hmmm i got a question i was looking through the code and here https://github.com/Lemonszz/Corvus/blob/master/src/main/java/party/lemons/corvus/gen/gaia/dimension/GaiaDimension.java 

what is this piece of code for

```java
	@SubscribeEvent
	public static void onSelectEntityToSpawn(LivingSpawnEvent.CheckSpawn event)
	{
		if(isInDim(event.getWorld()) && !(event.getEntity() instanceof EntityWitch) && event.getEntity() instanceof IMob && event.getSpawner() == null && event.getWorld().rand.nextInt(20) == 0)
		{
			event.setResult(Event.Result.DENY);
			EntityWitch witch = new EntityWitch(event.getWorld());
			witch.setPositionAndUpdate(event.getX(), event.getY(), event.getZ());

			event.getWorld().spawnEntity(witch);
		}
	}
```

as it looks like in the code ur spawning witches so that might be the issue

**Comment by Lemonszz** [comment]
Yeah- that's the code I'm suspicious of.

It's the case of working why that code is malfunctioning.

It works just fine on it's own, so the code is just fine. I assume it's some other mod calling that event incorrectly. (Not to pass blame, I'm willing to admit it's my fault if it comes to that :P)

I can probably change the way that that feature works to potentially bypass the issue, but I'd rather it be solved properly.

**Comment by vertex101** [comment]
>Yeah- that's the code I'm suspicious of. ...

@Lemonszz from what I seen doing my testing with a corrupted world there was over 1700 witches spawned in the Gaia dim and there was 3 to 4 witches stacked on top of each other

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by vertex101** [comment]
@Lemonszz in MCEdit doing a scan for the entities in the Gaia Dim this is what it shows in the console
![Entities](https://i.imgur.com/GlTtjh3.png)

and this is what it shows in the inspector for the entities
![Entities](https://i.imgur.com/CXG6TpU.png)
see how they are stacked in one location

so i wonder if in ur code if ur able to check if there is already an entity at that pos before spawning it and if there is an entity already there not to spawn one there

**Comment by MasterEnderman** [comment]
After some more investigation i think, this behavior is caused by trying to prevent witches from spawning in the gaia dimension. I tried "InControl!" as well as "Bad Mobs".

EDIT: started the pack again, without using any mod, that alters mobspawning, but the witches are still invisible / can't be killed. Furthermore i noticed that the player is unable to summon a witch.

**Comment by Lemonszz** [comment]
I'll change the way I'm spawning witches in the forest since it seems to be causing issues with *something somewhere*.
